tools/autograd/gen_autograd.py

```diff
@@ -57,13 +57,17 @@
     '_values': 'self',
     'indices': 'self',
     'values': 'self',
+    'view_as_real': 'self',
+    'view_as_complex': 'self',
     # sparse_coo ctor output should really be views of both indices and values,
     # but we only supports making as view of a single variable, and indices is
     # discrete anyways.
     # FIXME: clone indices on construction.
     'sparse_coo_tensor_with_dims_and_tensors': 'values',
 }
 
+VIEW_FUNCTIONS_WITH_DTYPE_CHANGE = ['view_as_real', 'view_as_complex']
```

Comment:
I think this is much more general than dtype change.
It should be all functions for which as_strided is not a good backward.

Reply:
hmm agreed but what are some other categories of functions for which as_strided would not be a good backward?

Reply:
I guess any function that changes metadata. Viewing between sparse and regular Tensors for example.